components/org.wso2.carbon.identity.application.authenticator.kakao/src/main/java/org/wso2/carbon/identity/application/authenticator/kakao/KakaoCustomAuthenticator.java

```diff
@@ -76,21 +123,251 @@ public List<Property> getConfigurationProperties() {
+                                JWT_TOKEN_VALIDATION_FAILED.getMessage(), e);
+                    }
+                }
+            } else {
+                String clientId = authenticatorProperties.get(CLIENT_ID);
+                String clientSecret = authenticatorProperties.get(CLIENT_SECRET);
+                String redirectUri = authenticatorProperties.get(CALLBACK_URL);
+                Boolean basicAuthEnabled = Boolean.parseBoolean(authenticatorProperties
+                        .get(Oauth2GenericAuthenticatorConstants.IS_BASIC_AUTH_ENABLED));
+                String code = getAuthorizationCode(request);
+                String tokenEP = getTokenEndpoint(authenticatorProperties);
+                token = getToken(tokenEP, clientId, clientSecret, code, redirectUri, basicAuthEnabled);
+            }
+
+            Boolean selfContainedTokenEnabled = Boolean.parseBoolean(authenticatorProperties
+                    .get(Oauth2GenericAuthenticatorConstants.SELF_CONTAINED_TOKEN_ENABLED));
+            String userInfo = getUserInfo(selfContainedTokenEnabled, token, authenticatorProperties);
+            buildClaims(context, userInfo);
+        } catch (ApplicationAuthenticatorException | MisconfigurationException e) {
+            String errorMessage = "Error while processing authentication response.";
+            throw new AuthenticationFailedException(errorMessage, e);
+        }
+    }
+
+    private boolean isTrustedTokenIssuer(AuthenticationContext context) {
+
+        ExternalIdPConfig externalIdPConfig = context.getExternalIdP();
+        if (externalIdPConfig == null) {
+            return false;
+        }
+
+        IdentityProvider externalIdentityProvider = externalIdPConfig.getIdentityProvider();
+        if (externalIdentityProvider == null) {
+            return false;
+        }
+
+        IdentityProviderProperty[] identityProviderProperties = externalIdentityProvider.getIdpProperties();
+        for (IdentityProviderProperty identityProviderProperty : identityProviderProperties) {
+            if (IdPManagementConstants.IS_TRUSTED_TOKEN_ISSUER.equals(identityProviderProperty.getName())) {
+                return Boolean.parseBoolean(identityProviderProperty.getValue());
+            }
+        }
+
+        return false;
+    }
+
+    private boolean isNativeSDKBasedFederationCall(HttpServletRequest request) {
+
+        return request.getParameter(ACCESS_TOKEN_PARAM) != null && request.getParameter(ID_TOKEN_PARAM) != null;
+    }
+
+    private AdditionalData getAdditionalData(AuthenticationContext context, boolean isNativeSDKBasedFederationCall) {
+
+        AdditionalData additionalData = new AdditionalData();
+
+        if (isNativeSDKBasedFederationCall) {
+            Map<String, String> additionalAuthenticationParams = new HashMap<>();
+            additionalData.setAdditionalAuthenticationParams(additionalAuthenticationParams);
```

Comment:
Is setting an empty additionalAuthenticationParams map necessary here?

Reply:
There was a missing data that needed to be added. This is addressed with https://github.com/wso2-extensions/identity-outbound-auth-kakao/pull/15/commits/8d9bd5da9b2a6348daffa96a0e58132d78f39d16

Reply:
Why did we remove this line?
additionalAuthenticationParams.put(STATE, context.getProperty(STATE).toString());

Reply:
This line is not removed, the `STATE` constant previously used was newly defined in the constants class. However this same value is available at Oauth2GenericAuthenticatorConstants.OAUTH2_PARAM_STATE. Therefore changed the line to reuse that constant and removed the newly created constant.